db/warehouse/migrate/20231120221840_colocated_warehouse_versions.rb

```diff
@@ -0,0 +1,17 @@
+class ColocatedWarehouseVersions < ActiveRecord::Migration[6.1]
+  def change
+    create_table :versions do |t|
+      t.string   :item_type, null: false
+      t.bigint   :item_id,   null: false
+      t.string   :event,     null: false
+      t.string   :whodunnit
+      t.jsonb    :object
+      t.jsonb    :object_changes
```

Comment:
Yay for JSON!

The client audit history feature currently calls [client.versions](https://github.com/greenriver/hmis-warehouse/blob/0d11a228fcb9926a63f8e2e1987d1a6d9ccc67a1/drivers/hmis/app/graphql/types/hmis_schema/has_audit_history.rb#L33) to get the version history, and then [expects object_changes to be YAML](https://github.com/greenriver/hmis-warehouse/blob/0d11a228fcb9926a63f8e2e1987d1a6d9ccc67a1/drivers/hmis/app/graphql/types/base_audit_event.rb#L64). So I think this would break that feature.

If that's right, we need to update the code to expect the changes to be JSON. Hopefully that is simple.

How difficult do you think it would be to migrate over the version records just for this scope `GrPaperTrail::Version.where(item_type: 'Hmis::Hud::Client')`? Client audit history is the only thing that's currently visible in the HMIS application, and I think we want to keep that history from the past couple months - but if it's a huge lift, we can discuss options, I'd just want to talk to customers first. It's also something we could migrate later on -- including maybe for other record types - if we need to.

Reply:
> If that's right, we need to update the code to expect the changes to be JSON. Hopefully that is simple.

Good catch! Yeah it doesn't look hard, I can do that here.

> How difficult do you think it would be to migrate over the version records just for this scope GrPaperTrail::Version.where(item_type: 'Hmis::Hud::Client')?

It shouldn't be a big lift, I think I can add a rake task for that in this PR

Reply:
Added a one-time rake task for this:
```sh
rails driver:hmis:migrate_versions_20231121
```

Reply:
Does it make sense to migrate the existing app table from yaml to json?  I'm guessing that'd be _hours_ of processing, so maybe not, but having them in two different formats seems like a future recipe for confusion.

Reply:
Yeah that seems like a good idea. I assume there's millions of rows in the existing tables so we probably need a migration strategy that's a bit more involved, recommend we make another issue for that. 

We may want to benchmark conversion so we can ball-park estimate how long the migration will take for a given db (hours, days, weeks?)

The paper trail gem suggests something like the following. Does this seem reasonable?
* rename existing object cols out of the way (object -> object_yml) and add new json-typed cols for object and object_changes
* paper trail should start using the new json cols going forward but will temporarily lose access to old object data
* Add a resumable task that pulls a batch of versions and migrates them to yaml. Scope would be something like `where object_yml IS NOT NULL AND object IS NULL`. This could be parallelized if needed
* Could also take the opportunity to fix other fields (whodunnit)
* Finally remove the object_yml and object_changes_yml once all rows are migrated